azure-pipelines-templates/device-test.yml

```diff
@@ -0,0 +1,156 @@
+parameters:
+  - name: appComponents
+    displayName: "Component name"
+    type: object
+  - name: unitTestRunsettings
+    displayName: |
+      Usage: unitTestRunsettings,unitTestDll
+      Example:
+        - nano.runsettings,bin/Release/NFUnitTest.dll
+        - Repo/nano.runsettings,Repo/bin/Release/NFUnitTest.dll
+      Unit test runsettings file. As default, nano.runsettings which will be represent $(System.DefaultWorkingDirectory)/Artifacts/nano.runsettings.
+      Unit test DLL. As default, bin/release/NFUnitTest.dll which will be represent $(System.DefaultWorkingDirectory)/Artifacts/bin/release/NFUnitTest.dll.
+    type: object
+    default: 
+      - 'nano.runsettings,bin/Release/NFUnitTest.dll'
+  - name: MaxRetries
+    displayName: "Maximum number of retries"
+    type: number
+    default: 5
+
+stages:
+  - stage: Test_${{ replace(parameters.appComponents, '-', '_') }}
+    dependsOn: Build
+    displayName: Test for ${{ parameters.appComponents }}
+
+    pool:
+      name: TestStream
+      demands: 
+        - ${{ parameters.appComponents }}
+
+    jobs:
+      - job:
+        displayName: 'Test job for ${{ parameters.appComponents }}'
+
+        steps:
+        - task: DownloadPipelineArtifact@2
+          displayName: 'Download Pipeline Artifact'
+          inputs:
+            path: '$(System.DefaultWorkingDirectory)'
+
+        - task: DotNetCoreCLI@2
+          displayName: 'Install nanoff'
+          inputs:
+            command: custom
+            custom: tool
+            arguments: 'install nanoff --tool-path /azp/tools'
```

Comment:
Why do we need to install to a specific location?
Unless there is a good reason for it, we should be using the default location.
In case the tool is already installed this will be quicker.

Reply:
There is a good reason: rights to install only in this directory. For security reasons, the only directory where anything can be installed is /azp directory. The others, the agent user do not have write access.

Reply:
> In case the tool is already installed this will be quicker.

It's the same, the location doesn't change. If the latest version is already installed, it's just a check and that's it.

Reply:
Got it! I was thinking on situations that the container could have this already installed for some reason like a pre-installed image with the tool or the test framework.